src/components/DepartmentForm/DepartmentForm.tsx

```diff
@@ -1,13 +1,63 @@
-import React from 'react';
-import { Form, Input } from 'reactstrap';
-
+import React, { useCallback, useEffect, useRef } from 'react';
+import { useDispatch } from 'react-redux';
+import { Form, Input, Button, Row } from 'reactstrap';
+import { Department } from '../../types/components/departments';
+import { AppThunkDispatch } from '../../types/store/appThunkTypes';
+import { closeModal } from '../../store/actionCreators/modalsActionCreators';
+import { addDepartmentThunk } from '../../store/actionCreators/departmentsActionCreators';
 import './DepartmentForm.scss';
 
 const DepartmentForm: React.FC = () => {
+  const nameRef = useRef<HTMLInputElement | null>(null);
+  const descritpionRef = useRef<HTMLInputElement | null>(null);
```

Comment:
o_O
all form elements should have value and onChange properties. Use state to manipulate that